zcash_client_backend/src/data_api/testing/pool.rs

```diff
@@ -821,85 +842,20 @@ pub fn send_multi_step_proposed_transfer<T: ShieldedPoolTester, DSF>(
-    // one built manually) have been mined yet. So, the range of address indices
-    // that are safe to reserve is still 0..20, and we have already reserved 12
-    // addresses, so trying to reserve another 9 should fail.
-    reservation_should_fail(&mut st, 9, 20);
-    reservation_should_succeed(&mut st, 8);
-    reservation_should_fail(&mut st, 1, 20);
-
-    // Now mine the transaction with the ephemeral output at index 1.
-    // We already reserved 20 addresses, so this should allow 2 more (..22).
-    // It does not matter that the transaction with ephemeral output at index 0
-    // remains unmined.
-    let (h, _) = st.generate_next_block_including(txids1.head);
-    st.scan_cached_blocks(h, 1);
-    reservation_should_succeed(&mut st, 2);
-    reservation_should_fail(&mut st, 1, 22);
-
-    // Mining the transaction with the ephemeral output at index 0 at this point
-    // should make no difference.
-    let (h, _) = st.generate_next_block_including(txids0.head);
-    st.scan_cached_blocks(h, 1);
-    reservation_should_fail(&mut st, 1, 22);
-
-    // Now mine the transaction with the ephemeral output at index 10.
-    let tx = build_result.transaction();
-    let tx_index = 1;
-    let (h, _) = st.generate_next_block_from_tx(tx_index, tx);
-    st.scan_cached_blocks(h, 1);
-
-    // The above `scan_cached_blocks` does not detect `tx` as interesting to the
-    // wallet. If a transaction is in the database with a null `mined_height`,
-    // as in this case, its `mined_height` will remain null unless either
-    // `put_tx_meta` or `set_transaction_status` is called on it. The former
-    // is normally called internally via `put_blocks` as a result of scanning,
-    // but not for the case of a fully transparent transaction. The latter is
-    // called by the wallet implementation in response to processing the
-    // `transaction_data_requests` queue.
-
-    // The reservation should fail because `tx` is not yet seen as mined.
-    reservation_should_fail(&mut st, 1, 22);
-
-    // Simulate the wallet processing the `transaction_data_requests` queue.
-    let tx_data_requests = st.wallet().transaction_data_requests().unwrap();
-    assert!(tx_data_requests.contains(&TransactionDataRequest::GetStatus(tx.txid())));
-
-    // Respond to the GetStatus request.
-    st.wallet_mut()
-        .set_transaction_status(tx.txid(), TransactionStatus::Mined(h))
-        .unwrap();
-
-    // We already reserved 22 addresses, so mining the transaction with the
-    // ephemeral output at index 10 should allow 9 more (..31).
-    reservation_should_succeed(&mut st, 9);
-    reservation_should_fail(&mut st, 1, 31);
-
-    let newest_known_addrs = st
-        .wallet()
-        .get_known_ephemeral_addresses(account_id, None)
-        .unwrap();
-    assert_eq!(newest_known_addrs.len(), (GAP_LIMIT as usize) + 31);
-    assert!(newest_known_addrs.starts_with(&known_addrs));
-    assert!(newest_known_addrs[5..].starts_with(&newer_known_addrs));
```

Comment:
I'm a bit concerned about loss of coverage here.

Reply:
We should look at this together; it was my conclusion that some of what was being tested here was not just implementation-dependent, but it was verifying behavior that I considered to be a bug.

Reply:
I'm happy with this now that the `gap_limits()` test has been added.